biogeophys/FatesLeafBiophysParamsMod.F90

```diff
@@ -0,0 +1,325 @@
+         dimension_names=dim_names, lower_bounds=dim_lower_bound)
+
+    name = 'fates_leaf_jmaxha'
+    call fates_params%RegisterParameter(name=name, dimension_shape=dimension_shape_1d, &
+         dimension_names=dim_names, lower_bounds=dim_lower_bound)
+
+    name = 'fates_leaf_vcmaxhd'
+    call fates_params%RegisterParameter(name=name, dimension_shape=dimension_shape_1d, &
+         dimension_names=dim_names, lower_bounds=dim_lower_bound)
+
+    name = 'fates_leaf_jmaxhd'
+    call fates_params%RegisterParameter(name=name, dimension_shape=dimension_shape_1d, &
+         dimension_names=dim_names, lower_bounds=dim_lower_bound)
+
+    name = 'fates_leaf_vcmaxse'
+    call fates_params%RegisterParameter(name=name, dimension_shape=dimension_shape_1d, &
+         dimension_names=dim_names, lower_bounds=dim_lower_bound)
+
+    name = 'fates_leaf_jmaxse'
+    call fates_params%RegisterParameter(name=name, dimension_shape=dimension_shape_1d, &
+         dimension_names=dim_names, lower_bounds=dim_lower_bound)
+
+    return
+  end subroutine LeafBiophysRegisterParams
+
+  ! =====================================================================================
+
+  subroutine LeafBiophysReceiveParams(fates_params)
+
+
+    class(fates_parameters_type), intent(inout) :: fates_params
+    real(r8), allocatable :: tmpreal(:)  ! Temporary variable to hold floats
+    real(r8)              :: tmpscalar
+    character(len=param_string_length) :: name
+
+    
+    !name = 'fates_leaf_theta_cj_c3'
+    !call fates_params%RetrieveParameter(name=name, &
+    !     data=lb_params%theta_cj_c3)
+
+    !name = 'fates_leaf_theta_cj_c4'
+    !call fates_params%RetrieveParameter(name=name, &
+    !     data=lb_params%theta_cj_c4)
+
+    name = 'fates_daylength_factor_switch'
+    call fates_params%RetrieveParameter(name=name, &
+         data=tmpscalar)
+    lb_params%dayl_switch = nint(tmpscalar)
+
+    name = 'fates_leaf_stomatal_model'
+    call fates_params%RetrieveParameter(name=name, &
+         data=tmpscalar)
+    lb_params%stomatal_model = nint(tmpscalar)
+
+    name = 'fates_leaf_stomatal_assim_model'
+    call fates_params%RetrieveParameter(name=name, &
+         data=tmpscalar)
+    lb_params%stomatal_assim_model = nint(tmpscalar)
+
+    name = 'fates_leaf_photo_tempsens_model'
+    call fates_params%RetrieveParameter(name=name, &
+         data=tmpscalar)
+    lb_params%photo_tempsens_model = nint(tmpscalar)
+
+    name = 'fates_leaf_c3psn'
+    call fates_params%RetrieveParameterAllocate(name=name, &
+         data=tmpreal)
+    allocate(lb_params%c3psn(size(tmpreal,dim=1)))
+    call ArrayNint(tmpreal,lb_params%c3psn)
+    deallocate(tmpreal)
+
+    ! Placeholder (enable for API 37, RGK)
+    !name = 'fates_leaf_stomatal_btran_model'
+    !call fates_params%RetrieveParameterAllocate(name=name, &
+    !     data=tmpreal)
+    allocate(lb_params%stomatal_btran_model(size(tmpreal,dim=1)))
```

Comment:
Maybe this will be resolved when it is enabled, but for now, isn't size(tmpreal,dim=1) undefined because it has been just deallocated in line 190?